### PR TITLE
rml: init at 1.09.07

### PR DESCRIPTION
**Pull request by wegank**
###### Description of changes

ReactiveML is a general purpose programming language extended with concurrency constructs to ease the programming of interactive systems.

https://rml.lri.fr

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/rml/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+stdenv.mkDerivation rec {
+  pname = "rml";
+  version = "1.09.07";
+
+  src = fetchFromGitHub {
+    owner = "reactiveml";
+    repo = pname;
+    rev = "${pname}-${version}-2021-07-26";
+    sha256 = "sha256-UFqXQBeIQMSV4O439j9s06p1hh7xA98Tu79FsjK9PIY=";
```
Comment (azahi) [inline pkgs/development/compilers/rml/default.nix:15]:
```suggestion
    hash = "sha256-UFqXQBeIQMSV4O439j9s06p1hh7xA98Tu79FsjK9PIY=";
```

Comment (ocfox) [reply pkgs/development/compilers/rml/default.nix:15]: Is this the new recommendation style?

Comment (wegank) [reply pkgs/development/compilers/rml/default.nix]: Done. Also NixOS/rfcs#131.

**Comment by azahi** [comment]
Result of `nixpkgs-review pr 200458` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>rml</li>
  </ul>
</details>

**Comment by azahi** [comment]
Looks like manual pages are not being installed. I think we should add them; here's the relevant Makefile [part](https://github.com/reactiveml/rml/blob/master/Makefile#L38) that seems to be lost. 

Also, it would be nicer if we could split outputs to `[ "out" "lib" "doc" ]`, but that's entirely optional.

**Comment by wegank** [comment]
@azahi There seems to be [no man page](https://github.com/reactiveml/rml/tree/master/man) to install.

**Comment by azahi** [comment]
> @azahi There seems to be [no man page](https://github.com/reactiveml/rml/tree/master/man) to install.

LOL. Alright :)

**Review by azahi — APPROVED** [review]

File: pkgs/development/compilers/rml/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  src = fetchFromGitHub {
+    owner = "reactiveml";
+    repo = pname;
+    rev = "${pname}-${version}-2021-07-26";
+    hash = "sha256-UFqXQBeIQMSV4O439j9s06p1hh7xA98Tu79FsjK9PIY=";
+  };
+
+  buildInputs = with ocamlPackages; [
+    ocaml
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/rml/default.nix:19]: Isn't ocaml run from PATH? Then it goes to nativeBuildInputs

Comment (wegank) [reply pkgs/development/compilers/rml/default.nix:19]: Done.

**Pull request merged**